### PR TITLE
Proofs

### PR DESCRIPTION
**Pull request by artyom-yurin**
### Description of the Change
- Added Seal functions:
VerifySeal
SealCommitPhase 1 and 2
Unseal
UnsealRange
- Added util functions:
FinalizeTicket
GetGpuDevices
GetPoStVersion
GetSealVersion
ClearCache
- Removed ParamProvider
- Removed bls submodule(Now everything works with filecoinf-ffi)
- Connected VerifySeal and VerifyPost to runtime

### Benefits
Actual proof version

### Possible Drawbacks 
None

### Usage Examples or Tests
`proofs_test::Lifecycle`

### CODE REVIEW

**Comment by eigendude** [comment]
Does this mean the Rust BLS signature library is no longer needed?

I believe filecoin-ffi is the only rust dependency left. I'm a bit confused, ffi is an interface to other code, but what code does filecoin-ffi interface with? or is it used to expose fuhon calls to outside code?

**Comment by artyom-yurin** [comment]
> Does this mean the Rust BLS signature library is no longer needed?

Yes, it does

> I believe filecoin-ffi is the only rust dependency left. I'm a bit confused, ffi is an interface to other code, but what code does filecoin-ffi interface with? or is it used to expose fuhon calls to outside code?

filecoin-ffi is a wrapper over [bls](https://github.com/filecoin-project/bls-signatures.git) and [rust-fil-proofs](https://github.com/filecoin-project/rust-filecoin-proofs-api.git)

**Comment by artyom-yurin** [comment]
Does this mean the Rust BLS signature library is no longer needed?

Yes, it does

I believe filecoin-ffi is the only rust dependency left. I'm a bit confused, ffi is an interface to other code, but what code does filecoin-ffi interface with? or is it used to expose fuhon calls to outside code?

filecoin-ffi is a wrapper over [bls](https://github.com/filecoin-project/bls-signatures.git) and [rust-filecoin-proofs-api](https://github.com/filecoin-project/rust-filecoin-proofs-api.git).

@eigendude

**Comment by eigendude** [comment]
> filecoin-ffi is a wrapper over [bls](https://github.com/filecoin-project/bls-signatures.git) and [rust-filecoin-proofs-api](https://github.com/filecoin-project/rust-filecoin-proofs-api.git).

I see, and filecoin-ffi pulls in rust-filecoin-proofs-api [here](https://github.com/filecoin-project/filecoin-ffi/blob/master/rust/Cargo.toml#L33)?

Are there plans to replace this last remaining rust dependency? I heard that replacing bls signatures was low priority, and yet here we are :) I ask because dropping rust entirely makes embedding fuhon 50% easier, though I was looking forward to the challenge of embedded rust.

**Comment by artyom-yurin** [comment]
Yes, it was a low priority. But new version of proofs had an intersection with old bls lib and conflicted with them. So that was decided to remove the old version.

About dropping this dependency, I think that's the answer is no. This lib will be updated later

**Comment by artyom-yurin** [comment]
@zuiris

File: core/primitives/address/CMakeLists.txt [hunk 1]
```diff
@@ -6,6 +6,7 @@ add_library(address
     address_codec.cpp
         ../piece/piece.hpp ../piece/piece.hpp)
```
[the remark this answers is not in the document]
Comment (artyom-yurin) [reply core/primitives/address/CMakeLists.txt]: Fixed

**Pull request merged**